gtda/graphs/geodesic_distance.py

```diff
@@ -52,18 +77,26 @@ class GraphGeodesicDistance(BaseEstimator, TransformerMixin, PlotterMixin):
 
     See also
     --------
-    TransitionGraph, KNeighborsGraph, gtda.homology.VietorisRipsPersistence
+    TransitionGraph, KNeighborsGraph
 
     """
 
-    def __init__(self, n_jobs=None):
+    def __init__(self, n_jobs=None, directed=False, unweighted=False,
+                 method='auto'):
         self.n_jobs = n_jobs
+        self.directed = directed
+        self.unweighted = unweighted
+        self.method = method
 
     def _geodesic_distance(self, X):
-        X_distance = graph_shortest_path(X)
-        X_distance[X_distance == 0] = np.inf  # graph_shortest_path returns a
-        # float64 array, so inserting np.inf does not change the type.
-        np.fill_diagonal(X_distance, 0)
+        if not issparse(X) and not isinstance(X, MaskedArray):
```

Comment:
What happens with boolean X? Is it passed directly to `shortest_path` and treated correctly?

Reply:
That's correct! Check e.g. the gist https://gist.github.com/ulupo/83cc82ce83379ebda8fdfe846d0c06a5

Reply:
The reason for converting to a masked array when the input is a dense one is to avoid the situation depicted in input cell 5 in that gist. Using `scipy` directly, any zeros will be interpreted as absent edges, and not as edges for which the cost is zero (which is more useful in our applications).